hugolib/hugo_sites_test.go

```diff
@@ -659,6 +670,106 @@ func TestChangeDefaultLanguage(t *testing.T) {
 	assertFileContent(t, "public/sect/doc2/index.html", true, "Single", "Hello")
 }
 
+func TestTableOfContentsInShortcodes(t *testing.T) {
+	testCommonResetState()
+
+	sites := createMultiTestSites(t, testSiteConfig{DefaultContentLanguage: "en"}, multiSiteTOMLConfigTemplate)
+
+	writeSource(t, "layouts/shortcodes/toc.html", tocShortcode)
+	writeSource(t, "content/post/simple.en.md", tocPageSimple)
+	writeSource(t, "content/post/withSCInHeading.en.md", tocPageWithShortcodesInHeadings)
+
+	cfg := BuildCfg{}
+
+	err := sites.Build(cfg)
+
+	if err != nil {
+		t.Fatalf("Failed to build sites: %s", err)
+	}
+
+	// Not checking the heading-ids because they are not generated by the ToC.
+	assertFileContentRegexp(t, "public/en/post/simple/index.html", true, tocPageSimpleExpected)
```

Comment:
You can turn off those funky heading IDs by setting plainIDAnchors = true or something -- it is set in another test, so search for it.